service/gcs/bridge/bridge.go

```diff
@@ -310,20 +289,23 @@ func (b *Bridge) ListenAndServe(bridgeIn io.ReadCloser, bridgeOut io.WriteCloser
 	go func() {
 		for req := range requestChan {
 			go func(r *Request) {
-				wr := &requestResponseWriter{
+				br := bridgeResponse{
 					header: &prot.MessageHeader{
 						Type: prot.GetResponseIdentifier(r.Header.Type),
 						ID:   r.Header.ID,
 					},
-					respChan: b.responseChan,
 				}
-				b.Handler.ServeMsg(wr, r)
-				if !wr.respWritten {
-					logrus.WithFields(logrus.Fields{
-						"message-type": r.Header.Type.String(),
-						"message-id":   r.Header.ID,
-					}).Error("opengcs::bridge - request didn't write response")
+				resp, err := b.Handler.ServeMsg(r)
+				if resp == nil {
+					resp = &prot.MessageResponseBase{
+						ActivityID: r.ActivityID,
+					}
 				}
+				resp.GetBase().ActivityID = r.ActivityID
+				setErrorForResponseBase(resp.GetBase(), err)
```

Comment:
Shouldn't we only do this if err != nil?

Reply:
I reverted the check in the function and did it here.